File: src/entities/panier.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

public class panier {
    private int idPanier;
    private String nom;
    private float prix;
    private int idProjet;
    private int qnt;


    public panier() {}

    public panier(int idPanier, String nom, float prix, int idProjet, int qnt) {
    this.idPanier = idPanier;
    this.nom = nom;
    this.prix = prix;
    this.idProjet = idProjet;
    this.qnt = qnt;
}

public int getQnt() {
    return qnt;
}

public void setQnt(int qnt) {
    this.qnt = qnt;
}

    public panier(int idPanier, String nom, float prix, int idProjet) {
        this.idPanier = idPanier;
        this.nom = nom;
        this.prix = prix;
        this.idProjet = idProjet;
    }

    public int getIdPanier() {
        return idPanier;
    }

    public void setIdPanier(int idPanier) {
        this.idPanier = idPanier;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public int getIdProjet() {
        return idProjet;
    }

    public void setIdProjet(int idProjet) {
        this.idProjet = idProjet;
    }
}


    
    

File: src/Entity/Panier.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
/**
 * Panier
 *
 * @ORM\Table(name="panier")
 * @ORM\Entity
 */
class Panier
{
    /**
     * @var int
     *@Groups({"paniers"})
     * @ORM\Column(name="idpanier", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idpanier;

    /**
     * @var string
     * *@Groups({"paniers"})
     * @ORM\Column(name="nom", type="string", length=255, nullable=false)
     */
    private $nom;

    /**
     * @var float
     * *@Groups({"paniers"})
     * @ORM\Column(name="prix", type="float", precision=10, scale=0, nullable=false)
     */
    private $prix;

    /**
     * @var int
     * *@Groups({"paniers"})
     * @ORM\Column(name="idprojet", type="integer", nullable=false)
     */
    private $idprojet;

    /**
     * @var int
     * *@Groups({"paniers"})
     * @ORM\Column(name="qnt", type="integer", nullable=false)
     */
    private $qnt;

    /**
     * @var int|null
     * *@Groups({"paniers"})
     * @ORM\Column(name="iduser", type="integer", nullable=true)
     */
    private $iduser;

    public function getIdpanier(): ?int
    {
        return $this->idpanier;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(?float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getIdprojet(): ?int
    {
        return $this->idprojet;
    }

    public function setIdprojet(?int $idprojet): self
    {
        $this->idprojet = $idprojet;

        return $this;
    }

    public function getQnt(): ?int
    {
        return $this->qnt;
    }

    public function setQnt(?int $qnt): self
    {
        $this->qnt = $qnt;

        return $this;
    }

    public function getIduser(): ?int
    {
        return $this->iduser;
    }

    public function setIduser(?int $iduser): self
    {
        $this->iduser = $iduser;

        return $this;
    }



}
